### ISSUE TITLE
ONNX Export Output has Incorrect Class Labels but Correct Box and Confidence

### ISSUE THREAD

**Issue by emirsahin1** [Deci-AI/super-gradients#2032]
### 💡 Your Question

I've trained a Yolonas Medium model on the COCO dateset. When I run the pytorch model, the results are as expected. However, when I export to ONNX, the class labels are not matching the detected object. The bounding boxes and confidence levels do match the results from the pytorch model though. Does anyone have any ideas as to why this could be?

Here is how I am exporting:
```python
model = models.get(Models.YOLO_NAS_M, num_classes=len(class_names), checkpoint_num_classes=len(class_names), checkpoint_path="./coco500M/ckpt_best.pth")
export_result = model.export("yolo_nas_new_MED.onnx", output_predictions_format=DetectionOutputFormatMode.FLAT_FORMAT, )
```

Here is my inference code:
```python
session = onnxruntime.InferenceSession("yolo_nas_new_MED.onnx", providers=["CUDAExecutionProvider"])
inputs = [o.name for o in session.get_inputs()]
outputs = [o.name for o in session.get_outputs()]
image = cv2.resize(image, (640, 640))
image_bchw = np.transpose(np.expand_dims(image, 0), (0, 3, 1, 2))
result = session.run(outputs, {inputs[0]: image_bchw})
```

### Versions

[pip3] numpy==1.23.0
[pip3] onnx==1.15.0
[pip3] onnx-graphsurgeon==0.3.27
[pip3] onnxruntime==1.15.0
[pip3] onnxsim==0.4.36
[pip3] torch==2.2.1
[pip3] torchmetrics==0.8.0
[pip3] torchvision==0.17.1

**Comment by emirsahin1** [comment]
Apologies, silly mistake on my end. The ordering of the labels I used in training was different than the one's found in super_gradient's COCO_DETECTION_CLASSES_LIST.
